Automattic/jetpack
# Subscriptions: update link to Subscribers page in Jetpack dashboard

## Proposed changes:

We currently link to an old Calypso page when folks want to see their subscribers after enabling the Subscriptions. We can do better than that.

In the past few months, 2 initiatives introduced 2 new screens that can be better than the existing one:

1. In Calypso, the `/subscribers` page includes more info and options than the existing `/people/subscribers` page.
2. In wp-admin (when the Stats feature is enabled, and when the new "Odyssey" stats experience is enabled), the Stats page includes a "Subscribers" page with extra info.

In this PR, we'll switch from the legacy to one of the 2 new options above, depending on availability.

I'm also updating the wording, but I'm not sure it's good enough at the moment:

<img width="519" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/27e3411b-9ca2-4829-a700-4ab8b42f0aa5">


### Other information:


## Jetpack product discussion

* p1688643479440219/1688481163.954019-slack-C0299DMPG
* pbNhbs-7bW-p2

## Does this pull request change what data or activity we track or use?

* The tracking links have been updated to take the new link options into account.

## Testing instructions:

* Your site will need to be connected to WordPress.com, and the Subscriptions module must be enabled.
* Go to Jetpack > Settings > Discussion
* Ensure the link to view subscribers works. It should be updated depending on your site settings
    * Is the Stats module enabled (you can disable it under Jetpack > Settings > Module)
    * Is the new Stats experience enabled (you can check that under Jetpack > Settings > Traffic
